frontend/views/pages/PendingApproval.vue

```diff
@@ -12,18 +13,37 @@ div
 </template>
 
 <script>
+import sbp from '@sbp/sbp'
 import { mapGetters } from 'vuex'
+import GroupWelcome from '@components/GroupWelcome.vue'
 import SvgInvitation from '@svgs/invitation.svg'
+import { PENDING_TO_WELCOME } from '@utils/events'
 
 export default ({
   name: 'PendingApproval',
   components: {
-    SvgInvitation
+    SvgInvitation,
+    GroupWelcome
+  },
+  data () {
+    return {
+      ephemeral: {
+        groupJoined: false
+      }
+    }
   },
   computed: {
     ...mapGetters([
       'groupSettings'
     ])
+  },
+  mounted () {
+    sbp('okTurtles.events/on', PENDING_TO_WELCOME, () => {
```

Comment:
Ricardo says I could watch for `!state.profiles?.[username]` where `state` is the group state

Reply:
we can create a computed property that looks at the specific group state by saving the currentGroupId when mounted